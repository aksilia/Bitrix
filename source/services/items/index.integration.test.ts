import Bitrix from '../../bitrix'

const WEBHOOK_URL = process.env['WEBHOOK_URL']

if (!WEBHOOK_URL) {
  throw Error('Integration tests require environmental variable `WEBHOOK_URL` to be set')
}

const { items } = Bitrix(WEBHOOK_URL)

describe('Items', () => {
  describe('fields', () => {
    it('should get list', async () => {
      const { result } = await items.list({entityTypeId: "167"})
      expect(result).toMatchSnapshot()
    })
  })
})
