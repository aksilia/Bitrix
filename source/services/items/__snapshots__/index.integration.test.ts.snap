// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Items fields should get list 1`] = `
Object {
  "assignedById": 166,
  "begindate": "2021-08-02T01:00:00+02:00",
  "categoryId": 1,
  "closedate": "2022-07-14T01:00:00+02:00",
  "companyId": 39677,
  "contactId": 0,
  "contactIds": Array [],
  "createdBy": 166,
  "createdTime": "2021-08-02T10:01:58+02:00",
  "currencyId": "EUR",
  "entityTypeId": 167,
  "id": 8,
  "isManualOpportunity": "Y",
  "lastActivityBy": 166,
  "lastActivityTime": "2021-08-02T10:01:58+02:00",
  "movedBy": 166,
  "movedTime": "2022-07-14T10:26:16+02:00",
  "mycompanyId": 39617,
  "observers": Array [],
  "opened": "N",
  "opportunity": 100000,
  "parentId2": 1101,
  "previousStageId": "DT167_1:3",
  "sourceDescription": "",
  "sourceId": "",
  "stageId": "DT167_1:SUCCESS",
  "taxValue": 0,
  "title": "RPS SRL_Microcredito",
  "ufCrm1_1627033790662": Array [],
  "ufCrm1_1627034673": Array [],
  "ufCrm1_1627034856986": "100000|EUR",
  "ufCrm1_1627035007321": null,
  "ufCrm1_1627035033920": null,
  "ufCrm1_1627035078776": null,
  "ufCrm1_1627035129935": null,
  "ufCrm1_1627035295039": Array [],
  "ufCrm1_1627035405": Array [],
  "ufCrm1_1627035465184": "",
  "ufCrm1_1627035899118": "finanziamento a tasso agevolato",
  "ufCrm1_1627035924480": "",
  "ufCrm1_1627296881141": null,
  "ufCrm1_1639589129938": null,
  "ufCrm1_1639589192768": null,
  "ufCrm1_1639589418612": null,
  "ufCrm1_1639647241": null,
  "ufCrm1_1639741414819": null,
  "ufCrm1_1648715172561": 2454,
  "ufCrm1_1654690782554": "Microcredito",
  "ufCrm1_1654690864": Array [],
  "updatedBy": 166,
  "updatedTime": "2022-07-14T10:26:16+02:00",
  "utmCampaign": null,
  "utmContent": null,
  "utmMedium": null,
  "utmSource": null,
  "utmTerm": null,
  "webformId": 0,
  "xmlId": "",
}
`;
